src/main/scala/com/microsoft/hyperspace/Hyperspace.scala

```diff
@@ -175,6 +176,10 @@ object Hyperspace extends ActiveSparkSession {
       // the one HyperspaceContext is using because Hyperspace depends on the
       // session's properties such as configs, etc.
       context.set(new HyperspaceContext(spark))
+      if (!ctx.spark.equals(spark)) {
```

Comment:
@imback82 I'm thinking of this change & what can we do about this. 
I think this is the problem - we access using the active spark session when optimizing query plan, but it can be different from the session which is used to create Hyperspace object.
So.. how about 
1) not resetting hyperspace & just using the previous one even with different active spark session. 
2) if a user wants to use another configs/spark session, they need to redefine Hyperspace object. 

I think this is clearer behavior because we do `val hs = new Hyperspace(spark)`

Reply:
> 1. not resetting hyperspace & just using the previous one even with different active spark session.

Will it be ok to silently use a different active session?